File: src/codes/Users_Manual/Tutorial/move_focus/soft.cs
~using AUTD3Sharp;
~using AUTD3Sharp.Utils;
~using AUTD3Sharp.Link;
~using AUTD3Sharp.Gain;
~using AUTD3Sharp.Modulation;
~using static AUTD3Sharp.Units;
~using var autd = Controller.Open([new AUTD3(pos: Point3.Origin, rot: Quaternion.Identity)], new Nop());
autd.Send(new Sine(
    freq: 150u * Hz,
    option: new SineOption()
));

var center = autd.Center() + new Vector3(0.0f, 0.0f, 150.0f);
while (true)
{
    autd.Send(new Focus(
        pos: center + new Vector3(20.0f, 0.0f, 0.0f),
        option: new FocusOption()
    ));

    Thread.Sleep(1000);

    autd.Send(new Focus(
        pos: center - new Vector3(20.0f, 0.0f, 0.0f),
        option: new FocusOption()
    ));

    Thread.Sleep(1000);
}

File: src/codes/Users_Manual/Tutorial/move_focus/soft.py
~import time
~import numpy as np
~from pyautd3 import AUTD3, Controller, Focus, FocusOption, Hz, Sine, SineOption
~from pyautd3.link.nop import Nop
~autd = Controller.open([AUTD3(pos=[0.0, 0.0, 0.0], rot=[1, 0, 0, 0])], Nop())
autd.send(
    Sine(
        freq=150 * Hz,
        option=SineOption(),
    )
)

center = autd.center() + np.array([0.0, 0.0, 150.0])
while True:
    autd.send(
        Focus(
            pos=center + np.array([20.0, 0.0, 0.0]),
            option=FocusOption(),
        ),
    )

    time.sleep(1)

    autd.send(
        Focus(
            pos=center - np.array([20.0, 0.0, 0.0]),
            option=FocusOption(),
        ),
    )

    time.sleep(1)